pkg/workflow_execution.go

```diff
@@ -1463,3 +1463,28 @@ func (c *Client) getWorkflowExecutionAndTemplate(namespace string, uid string) (
 
 	return
 }
+
+func (c *Client) DeleteWorkflowExecution(namespace, uid string) error {
+	err := c.ArgoprojV1alpha1().Workflows(namespace).Delete(uid, nil)
+	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			return nil
+		}
+		return err
+	}
+	return nil
+}
+
+func (c *Client) DeleteWorkflowExecutionDb(workflowUid string) error {
```

Comment:
I believe this needs a namespace to be uniquely identified?

Reply:
Also, I think this should be archiving it and not actually deleting it? Might be wrong.

Reply:
You are right about namespace.

Regarding archive vs delete, I added a separate method to archive.

I ended up removing this function because it's not being used anywhere.
- Plus, the sql inside doesn't save much code (per our discussion this morning); so it shouldn't be a function anyway.